.github/workflows/build_and_test.yml

```diff
@@ -53,6 +53,9 @@ on:
       codecov_token:
         description: The upload token of codecov.
         required: false
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
This actually was proposed before https://github.com/apache/spark/pull/40822. Do you remember why it was closed, @yaooqinn ?

Reply:
It was correctly cancelled - https://github.com/gaogaotiantian/spark/runs/55141811395